src/PlaneWaveBasis.jl

```diff
@@ -470,3 +480,65 @@ function PlaneWaveBasis(basis::PlaneWaveBasis; use_symmetry)
                                [[identity_symop()] for _ in 1:length(kcoords)];
                                fft_size=basis.fft_size)
 end
+
+
+"""
+Gather the distributed k-Point data on the master process and return
+it as a `PlaneWaveBasis`. On the other (non-master) processes `nothing` is returned.
+The returned object should not be used for computations and only to extract data
+for post-processing and serialisation to disk.
+"""
+function mpi_kgather(basis::PlaneWaveBasis)
+    # No need to allocate and setup a new basis object
+    mpi_nprocs(basis.comm_kpts) == 1 && return basis
+
+    # Gather k-Point info on master
+    kcoords = getproperty.(basis.kpoints, :coordinate)
+    kcoords = mpi_kgather(kcoords, basis)
+    ksymops = mpi_kgather(basis.ksymops, basis)
+
+    # Number of distinct k-Point coordinates is number of k-Points with spin 1
+    n_spinup_thisproc = count(isequal(1), getproperty.(basis.kpoints, :spin))
+    n_kcoords = mpi_sum(n_spinup_thisproc, basis.comm_kpts)
+
+    if isnothing(kcoords)  # i.e. master process
+        nothing
+    else
+        PlaneWaveBasis(basis.model,
+                       basis.Ecut,
+                       kcoords[1:n_kcoords],
+                       ksymops[1:n_kcoords],
+                       basis.symmetries;
+                       fft_size=basis.fft_size,
+                       kgrid=basis.kgrid,
+                       kshift=basis.kshift,
+                       variational=true,  # TODO hard-coded
```

Comment:
assert the incoming pwbasis was variational

Reply:
That's not easy to do, but I thought about adding a "variational" flag to pwbasis anyway.